crates/host/src/secrets.rs

```diff
@@ -138,60 +139,61 @@ impl Manager {
             .then(|secret_name| async move {
                 match self.config_store.get(&secret_name).await {
                     Ok(Some(secret)) => serde_json::from_slice::<SecretReference>(&secret)
-                        .map_err(|_| anyhow::anyhow!("failed to deserialize secret reference from config store, ensure {secret_name} is a secret reference and not configuration")),
+                        .map_err(|_| anyhow::anyhow!("failed to deserialize secret reference from config store, ensure {secret_name} is a secret reference and not configuration"))
+                        .map(|secret_ref| (secret_name.trim_start_matches(SECRET_PREFIX).to_string(), secret_ref)),
                     Ok(None) => bail!(
                         "Secret reference {secret_name} not found in config store"
                     ),
                     Err(e) => bail!(e),
                 }
             })
             // Retrieve the actual secret from the secrets backend
-            .and_then(|secret_ref| async move {
+            .and_then(|(secret_name, secret_ref)| async move {
                 let secrets_client = self
                     .get_or_create_secrets_client(&secret_ref.backend)
                     .await?;
-                let application = Application{
-                    name: application.cloned().unwrap_or_default(),
-                    policy: secret_ref.policy_properties.clone(),
-                };
                 let request = SecretRequest {
-                    name: secret_ref.key.clone(),
-                    version: secret_ref.version.clone(),
+                    name: secret_ref.key,
+                    version: secret_ref.version,
                     context: Context {
                         entity_jwt: entity_jwt.to_string(),
                         host_jwt: host_jwt.to_string(),
-                        application,
+                        application: Application {
+                            // We pass an empty string if the entity doesn't belong to an application
+                            name: application.cloned().unwrap_or_default(),
```

Comment:
What was the reasoning against using an `Option` here? Just that `Application` should 99% of the time have a `name`?

Reply:
Yeah originally it was optional, but now we changed it to be a struct. I think the application name here can still be optional and that will work well, but policy should always be there